lib/dl_api_lib/dl_api_lib/connector_availability/base.py

```diff
@@ -223,6 +283,43 @@ class ConnectorAvailabilityConfig(SettingsBase):
         env_var_converter=conn_type_set_env_var_converter,
         missing_factory=set,
     )
+    icon_src: ConnectorIconSrcConfig = attr.ib(default=None)
+
+    _icon_by_conn_type: dict[str, dict[str, Any]] = attr.ib(factory=dict)
+
+    def fill_icon_dict_by_conn_type(self) -> None:
+        if not self._icon_by_conn_type and self.icon_src is not None:
+            self._icon_by_conn_type = dict()
+            icon_src = self.icon_src.as_dict()
+            for conn in self._iter_connectors():
+                conn_type = conn.conn_type.value
+                self._icon_by_conn_type[conn_type] = dict()
+                self._icon_by_conn_type[conn_type]["type"] = icon_src["type"].value
+                self._icon_by_conn_type[conn_type]["conn_type"] = conn.conn_type.value
+                if icon_src["type"] == ConnectorIconSrcType.url:
+                    self._icon_by_conn_type[conn_type][ConnectorIconSrcType.url.value] = dict()
+                    self._icon_by_conn_type[conn_type][ConnectorIconSrcType.url.value][
+                        ConnectorIconRole.standard.value
+                    ] = icon_src["url_prefix"] + (conn.conn_type.value + ".svg")
+                    self._icon_by_conn_type[conn_type][ConnectorIconSrcType.url.value][
+                        ConnectorIconRole.nav.value
+                    ] = icon_src["url_prefix"] + (conn.conn_type.value + ".svg")
+                else:
+                    self._icon_by_conn_type[conn_type][ConnectorIconSrcType.data.value] = dict()
+                    self._icon_by_conn_type[conn_type][ConnectorIconSrcType.data.value][
+                        ConnectorIconRole.standard.value
+                    ] = ""
+                    self._icon_by_conn_type[conn_type][ConnectorIconSrcType.data.value][
+                        ConnectorIconRole.nav.value
+                    ] = ""
```

Comment:
This part is really hard to read. Isn't it mostly `self._icon_by_conn_type[conn_type] = deepcopy(icon_src)`? Otherwise, I don't get the point of `ConnectorIconSrcConfig.as_dict()`.